expression/builtin_time.go

```diff
@@ -2977,15 +2999,17 @@ func (du *baseDateArithmetical) subDuration(ctx sessionctx.Context, d types.Dura
 	if err != nil {
 		return types.ZeroDuration, true, err
 	}
+	// Adjust fsp as required by outer - always respect type inference.
+	retDur.Fsp = resultFsp
 	return retDur, false, nil
```

Comment:
Maybe replace this function body with:
`return addDuration(ctx, Neg(d), interval, unit, resultFsp)`?

Reply:
Nice catch. Will do.

Reply:
Addressed.

Reply:
Actually it is "interval" that needs to be neg-ed. Kind of tricky because interval is unparsed string. Looking for other way around.

Reply:
Addressed using composition.